core/parachain/validator/impl/parachain_observer.cpp

```diff
@@ -153,7 +153,8 @@ namespace kagome::observers {
 
       if (peer_state->get().collator_state) {
         logger_->warn("Peer is in collating state {}:{}", peer_id, para_id);
-        // return;
+        // TODO(iceseer): check that peer is not in collating state, or is in
+        // collating state with similar pubkey and parachain id.
```

Comment:
If this `todo` for future, please, create issue. Without that it will be forgotten.